spark/datadog_checks/spark/spark.py

```diff
@@ -577,7 +577,14 @@ def _set_metrics_from_json(self, tags, metrics_json, metrics):
             return
 
         for status, (metric_name, metric_type) in iteritems(metrics):
-            metric_status = metrics_json.get(status)
+            # Metrics defined with a dot `.` are exposed in a nested dictionary.
+            # {"foo": {"bar": "baz", "qux": "quux"}}
+            #   foo.bar -> baz
+            #   foo.qux -> quux
+            if '.' in status:
+                metric_status = metrics_json.get(status.split('.')[0], {}).get(status.split('.')[1])
```

Comment:
If you're sure that you will only get 2 items, this is more easy to understand:
```suggestion
                pt1, pt2 = status.split('.')
                metric_status = metrics_json.get(pt1, {}).get(pt_2)
```